src/froll.c

```diff
@@ -349,28 +369,27 @@ void frollsumExact(double *x, uint64_t nx, ans_t *ans, int k, double fill, bool
       }
       if (R_FINITE((double) w)) {
         ans->dbl_v[i] = (double) w;
-      } else {
+      } else if (ISNAN((double) w)) {
         if (!narm) {
           ans->dbl_v[i] = (double) w;
         }
-        truehasna = true;
+        truehasnf = true;
+      } else {
+        ans->dbl_v[i] = (double) w;
       }
```

Comment:
Remove first if since same as else?

Reply:
Here will fall +Inf and -Inf (non finite && non an NA[N]). If codecov doesn't complaint for this line then it must be reached by some tests.